test/minikube/minikube_rolling_upgrade_test.go

```diff
@@ -31,6 +32,119 @@ func verifyAllProcessesRunning(t *testing.T, namespaceName string, adminPod stri
 	}, 30*time.Second)
 }
 
+func TestAdminReadinessProbe(t *testing.T) {
+	// TODO: remove this whenever the image tested in nuodb-helm-charts CI
+	// supports 'nuocmd check server' (singular)
+	if os.Getenv("NUODB_DEV") != "true" {
+		t.Skip("'nuocmd check server' is not supported in released versions")
+	}
+
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	// create a two-server domain and induce a failure that makes it
+	// impossible to elect a leader, causing 'nuocmd check server
+	// --check-converged' to fail
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{
+		SetValues: map[string]string{
+			"admin.replicas": "2",
+		},
+	}, 2, "")
+	adminStatefulSet := helmChartReleaseName + "-nuodb-cluster0"
+	admin0 := adminStatefulSet + "-0"
+	admin1 := adminStatefulSet + "-1"
+
+	// make sure both Admin Pods become Ready
+	testlib.AwaitPodUp(t, namespaceName, admin0, 120*time.Second)
+	testlib.AwaitPodUp(t, namespaceName, admin1, 120*time.Second)
+
+	// make sure direct invocation of readinessprobe script succeeds on both
+	// Admin processes
+	options := k8s.NewKubectlOptions("", "", namespaceName)
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", admin0, "--", "readinessprobe")
+	require.NoError(t, err, "readinessprobe failed: %s", output)
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "--", "readinessprobe")
+	require.NoError(t, err, "readinessprobe failed: %s", output)
+
+	leader, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", admin0, "--", "bash", "-c",
+		"nuocmd show domain --server-format '{role} {id}' | sed -n 's/ *LEADER //p'")
+	require.NoError(t, err)
+	// delete Raft log on admin-0 and kill admin-0 so that it bootstraps a
+	// disjoint domain when it is restarted and refuses messages from
+	// admin-1
+	k8s.RunKubectl(t, options, "exec", admin0, "--", "rm", "-f", "/var/opt/nuodb/raftlog")
+	k8s.RunKubectl(t, options, "delete", "pod", admin0)
+	// also kill admin-1 if it was the leader so that a new leader has to be
+	// elected when it is restarted
+	if admin1 == strings.TrimSpace(leader) {
+		k8s.RunKubectl(t, options, "delete", "pod", admin1)
+	}
+
+	// make sure readinessprobe on admin-1 eventually fails, because there
+	// is no leader for it to converge with even if the backwards-compatible
+	// 'nuocmd check servers' (plural) succeeds; this verifies that the
+	// fallback behavior does not apply to actual failures of the readiness
+	// check
+	testlib.Await(t, func() bool {
+		// make sure 'nuocmd check servers' (plural) is successful, since it
+		// only checks that the --api-server is ACTIVE
+		output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "--", "nuocmd", "check", "servers")
```

Comment:
do you need the output here? I would use `RunKubectlE`

Reply:
Will do.

Reply:
This is no longer relevant now that the test has been updated to account for the new fallback behavior.